src/EditorFeatures/CSharpTest/ConvertForToForEach/ConvertForToForEachTests.cs

```diff
@@ -1006,6 +933,79 @@ void Test(string[] array)
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)]
+        public async Task TestMissingIfCollectionWrittenTo()
```

Comment:
note: do we actually want to change this behavior?  wasn't it nice that you coudl do this, and get a nice warning?  It may be that we need to tighten up the warning to be shown in more places.  But for a refactoring, isn't that preferred?